src/config/feature-flags.ts

```diff
@@ -0,0 +1,138 @@
+import LaunchDarkly, {LDClient, LDLogger, LDUser} from "launchdarkly-node-server-sdk";
+import crypto from "crypto";
+import {getLogger} from "./logger";
+import Logger from "bunyan";
+
+const logger = getLogger("feature-flags");
+
+/**
+ * Abstraction over the actual feature flag provider so we can
+ * multiple implementations.
+ */
+interface FeatureFlagResolver {
+	booleanValue(flag: string, jiraHost: string, defaultValue: boolean): Promise<boolean>;
+}
+
+/**
+ * A no-op feature flag resolver that always returns the default value.
+ */
+class NoopFeatureFlagResolver implements FeatureFlagResolver {
+	booleanValue(_: string, __: string, defaultValue: boolean): Promise<boolean> {
+		return Promise.resolve(defaultValue);
+	}
+}
+
+/**
+ * A feature flag resolver that resolves feature flags against the LaunchDarkly server.
+ * Uses the SHA-1 hash of the Jira Host URL to identify a user in LaunchDarkly (because
+ * we don't want to send the plain URL to LaunchDarkly). If you want to target specific
+ * Jira sites, you can create a SHA-1 hash on https://www.fileformat.info/tool/hash.htm.
+ */
+class LaunchdarklyFeatureFlagResolver implements FeatureFlagResolver {
+
+	private launchdarklyClient: LDClient;
+
+	async initWithLaunchdarklyKey(launchdarklyKey: string): Promise<void> {
+		try {
+			this.launchdarklyClient = LaunchDarkly.init(launchdarklyKey, {
+				logger: new LaunchDarklyLogger()
+			});
+			await this.launchdarklyClient.waitForInitialization();
+		} catch (e) {
+			throw new Error(`Failed to wait for LD Client initialization: ${e}`);
+		}
+	}
+
+	/**
+	 * Creates a LaunchDarkly user using the hash value of the user's Jira host URL.
+	 * We don't want to send the plain Jira host URL to LaunchDarkly, because it's
+	 * considered personal data, so we hash it instead.
+	 */
+	createLaunchDarklyUser(jiraHost: string): LDUser {
+		const hash = crypto.createHash("sha1");
+		hash.update(jiraHost)
+		const hashAsHex = hash.digest("hex");
+		return {
+			key: hashAsHex
+		}
+	}
+
+	booleanValue(flag: string, jiraHost: string, defaultValue: boolean): Promise<boolean> {
```

Comment:
Hence we not always having Jira context, should we have a method which doesn't accepts jiraHost and returns value for "default user"? 

For example, during webhook processing we don't have jira instance.

Reply:
I thought about adding such a function, but decided against it because I wanted to force the developer into the default of having a feature flag that is scoped via jiraHostUrl. 

During webhook processing, the maintenance window feature flag activates only after it knows which jiraHost the webhook belongs to.

Reply:
Ok, I guess we can add if we'll really need it in the future